### PR TITLE
charls: Use absolute target names

### PR DESCRIPTION
**Pull request by franramirez688**
This PR updates the target names for CMakeDeps via set_property.
Conan version required: 1.43
Conan feature related: conan-io/conan#10099

### CODE REVIEW

**Review by SpaceIm — APPROVED** [review]

File: recipes/charls/all/conanfile.py [hunk 1]
```diff
@@ -110,9 +110,8 @@ def package(self):
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "charls")
         self.cpp_info.set_property("cmake_target_name", "charls")
-        self.cpp_info.builddirs.append(self._module_subfolder)
-        self.cpp_info.set_property("cmake_build_modules", [self._module_file_rel_path])
```
Comment (ericLemanissier) [inline recipes/charls/all/conanfile.py:114]: this PR does not change target names, but removes `cmake_build_modules` property. Is it intended ? is the title inaccurate?

Comment (SpaceIm) [reply recipes/charls/all/conanfile.py:114]: yes it is intended, the module is a trick to provide non-namespaced imported target in the old `.names` logic, but it's useless for the new `cmake_target_name` logic.

Comment (ericLemanissier) [reply recipes/charls/all/conanfile.py:114]: so the intent of this PR is to remove some no-op code ?

Comment (SpaceIm) [reply recipes/charls/all/conanfile.py:114]: I think the intent is to remove in CCI any logic based on the previous logic of `set_property()`.

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**